File: api/WebApp.Domain/Entities/Priority.cs
namespace WebApp.Domain.Entities;

public sealed record Priority
{
    public PriorityId Id { get; init; }
    public Project Project { get; init; } = null!;
    public ProjectId ProjectId { get; init; }
    public string Name { get; init; } = null!;
    public string Color { get; init; } = null!;
    public string Rank { get; init; } = null!;
    public string? Description { get; init; }
}

File: www/src/lib/models/priority.ts
import type { Project } from './project';

export interface Priority {
    id: string;
    name: string;
    color: string;
    rank: string;
    category: PriorityCategory;
    description?: string;
    projectId: string;
    project: Project;
}

export enum PriorityCategory {
    Low = 'low',
    Medium = 'medium',
    High = 'high',
    Urgent = 'urgent',
}
